dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java

```diff
@@ -858,7 +848,7 @@ public void addToBeanClazzes(Class<?> clazz) {
 
     @Override
     public void createInjectionTargetsForJEEComponentClasses() throws CDIException {
-        Set<Class<?>> jeeComponentClasses = getJEEComponentClasses();
+        Set<Class<?>> jeeComponentClasses = new HashSet<>(getJEEComponentClasses());
```

Comment:
check if this change was really needed

Reply:
getJEEComponentClasses() is going to do a recursive search every time it's called. Surely we can do better than that. 

As an added bonus my change will mean that we can safely remove the new HashSet<> since I'm synchronising the thread there's no need for a safety hashset